languages/elixir/exercises/concept/multiple-clause-functions/.meta/design.md

```diff
@@ -0,0 +1,61 @@
+# Concept Exercise: multiple-clause-functions
+
+## Goal
+
+The goal for this exercise is to teach more about functions in elixir and how to use multiple functions clauses, guards
+
+## Learning objectives
+
+A student upon completion will:
+
+- Know what a guard is.
+- Know how to use a guard.
+- Know that guards return strict boolean values.
+- Know that guards meet certain guarantees to not mutate state.
+- Know that guards must be declared in a special form, and not all functions that return boolean values are guards.
+- Know how to write basic multiple-clause functions.
+- Know how to use guards to protect execution of a function.
+- Know how to use guards to perform basic control flow.
+- Know how to specify a default argument
+- Know that by specifying a default argument, it defines multiple arities for the function name
+
+## Out of scope
+
+- pattern matching
+
+## Concepts
+
+- `atoms`
+- `multiple-clause-functions`
+- `guards`
+- `default-parameters`
+
+## Prerequisites
+
+- `basics`
+- `booleans`
+- `conditionals`
+
+## Narrative
+
+This exercise is a fork from [F#'s Number Guessing Game](https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/pattern-matching/.docs/instructions.md)
+
+## Resources to refer to
+
+- <https://hexdocs.pm/elixir/master/Kernel.html#guards>
+- <https://elixir-lang.org/getting-started/modules-and-functions.html#default-arguments>
+- [multiple clauses in anon functions](https://til.hashrocket.com/posts/36c6d2684e-defining-multiple-clauses-in-an-anonymous-function)
+- <https://elixirschool.com/en/lessons/basics/functions/#guards>
+- <https://elixirschool.com/en/lessons/basics/functions/#default-arguments>
+
+## Representer
+
+No modifications at this time
+
+## Analyzer
+
+No analyzer features at this time.
```

Comment:
I think it will be pretty crucial to assert that people are actually using multiple function definitions and guards because it is very easy to do this exercise completely without them. Will that be possible?

Reply:
It would, I can add this in here.

Arguably most of the exercises can probably sabotaged with this behaviour, so there is probably room for things like this in every exercise.